tests/1_engine/rex_engine_unit_test/src/test_path.cpp

```diff
@@ -0,0 +1,525 @@
+#include "rex_test/rex_catch2.h"
+
+#include "rex_engine/filesystem/path.h"
+#include "rex_std/chrono.h"
+#include "rex_std/thread.h"
+#include "rex_std/iostream.h"
+
+#include <Windows.h>
+#include <ShlObj.h>
+
+TEST_CASE("Path Joining")
+{
+  CHECK(rex::path::join("D:", "foo", "bar") == "D:/foo/bar");
+  CHECK(rex::path::join("foo", "bar") == "foo/bar");
+  CHECK(rex::path::join("foo", "bar.txt") == "foo/bar.txt");
+}
+
+TEST_CASE("Change Extension")
+{
+  CHECK(rex::path::change_extension("foo.txt", ".bar") == "foo.bar");
+  CHECK(rex::path::change_extension("foo.txt", "") == "foo");
+  CHECK(rex::path::change_extension("foo", ".bar") == "foo.bar");
+  CHECK(rex::path::change_extension("foo", "") == "foo");
+
+  CHECK(rex::path::change_extension("bar/foo.txt", ".bar") == "bar/foo.bar");
+  CHECK(rex::path::change_extension("bar/foo.txt", "") == "bar/foo");
+  CHECK(rex::path::change_extension("bar/foo", ".bar") == "bar/foo.bar");
+  CHECK(rex::path::change_extension("bar/foo", "") == "bar/foo");
+}
+
+TEST_CASE("Dir Name")
+{
+  CHECK(rex::path::dir_name("foo.txt") == "");
+  CHECK(rex::path::dir_name("bar/foo.txt") == "bar");
+}
+
+TEST_CASE("Get Extension")
+{
+  CHECK(rex::path::extension("foo.txt") == ".txt");
+  CHECK(rex::path::extension("foo.txt.bar") == ".txt.bar");
+  CHECK(rex::path::extension("foo") == "");
+  
+  CHECK(rex::path::extension("bar/foo.txt") == ".txt");
+  CHECK(rex::path::extension("bar/foo.txt.bar") == ".txt.bar");
+  CHECK(rex::path::extension("bar/foo") == "");
+}
+
+TEST_CASE("Get Stem")
+{
+  CHECK(rex::path::stem("foo.txt") == "foo");
+  CHECK(rex::path::stem("foo.txt.bar") == "foo");
+  CHECK(rex::path::stem("foo") == "foo");
+
+  CHECK(rex::path::stem("bar/foo.txt") == "foo");
+  CHECK(rex::path::stem("bar/foo.txt.bar") == "foo");
+  CHECK(rex::path::stem("bar/foo") == "foo");
+}
+
+TEST_CASE("Absolute Path")
+{
+  rsl::string cwd = rex::path::cwd();
+
+  CHECK(rex::path::abs_path("foo.txt") == rex::path::join(cwd, "foo.txt"));
+  CHECK(rex::path::abs_path("foo.txt.bar") == rex::path::join(cwd, "foo.txt.bar"));
+  CHECK(rex::path::abs_path("foo") == rex::path::join(cwd, "foo"));
+
+  CHECK(rex::path::abs_path("bar/foo.txt") == rex::path::join(cwd, "bar", "foo.txt"));
+  CHECK(rex::path::abs_path("bar/foo.txt.bar") == rex::path::join(cwd, "bar", "foo.txt.bar"));
+  CHECK(rex::path::abs_path("bar/foo") == rex::path::join(cwd, "bar", "foo"));
+
+  CHECK(rex::path::abs_path("c:/bar/foo.txt") == "c:/bar/foo.txt");
+  CHECK(rex::path::abs_path("c:/bar/foo.txt.bar") == "c:/bar/foo.txt.bar");
+  CHECK(rex::path::abs_path("c:/bar/foo") == "c:/bar/foo");
+}
+
+TEST_CASE("Path Root")
+{
+  CHECK(rex::path::path_root("c:/bar/foo.txt") == "c:");
```

Comment:
Can you add a check for ./bar/foo.txt
Can you add a check for /bar/foo.txt
Can you add a check for ~/bar/foo.txt